packages/db/src/meta/process/run.ts

```diff
@@ -2,7 +2,7 @@ import { logger } from "@truffle/db/logger";
 const debug = logger("db:meta:process:run");
 
 import { promisify } from "util";
-import type { Provider } from "web3/providers";
+import type { Web3BaseProvider as Provider } from "web3-types";
```

Comment:
We shouldn't need to cast it to `Provider`, can you share the type error you were getting?